logstash-core/src/main/java/org/logstash/instrument/witness/EventsWitness.java

```diff
@@ -0,0 +1,205 @@
+
+/**
+ * Witness for events.
+ */
+final public class EventsWitness{
+
+    private LongCounter filtered;
+    private LongCounter out;
+    private LongCounter in;
+    private LongCounter duration;
+    private LongCounter queuePushDuration;
+    private final Snitch snitch;
+    private final Forgetter forgetter;
+    private boolean dirty; //here for passivity with legacy Ruby implementation
+
+    /**
+     * Constructor.
+     */
+    public EventsWitness() {
+        filtered = new LongCounter("filtered");
+        out = new LongCounter("out");
+        in = new LongCounter("in");
+        duration = new LongCounter("duration_in_millis");
+        queuePushDuration = new LongCounter("queue_push_duration_in_millis");
+        snitch = new Snitch(this);
+        forgetter = new Forgetter(this);
+        dirty = false;
+    }
+
+    /**
+     * Add to the existing duration
+     *
+     * @param durationToAdd the amount to add to the existing duration.
+     */
+    public void duration(long durationToAdd) {
+        duration.increment(durationToAdd);
+        dirty = true;
+    }
+
+    /**
+     * increment the filtered count by 1
+     */
+    public void filtered() {
+        filtered.increment();
+        dirty = true;
+    }
+
+    /**
+     * increment the filtered count
+     *
+     * @param count the count to increment by
+     */
+    public void filtered(long count) {
+        filtered.increment(count);
+        dirty = true;
+    }
+
+    /**
+     * Get a reference to associated forgetter.
+     *
+     * @return the associate {@link Forgetter}
+     */
+    public Forgetter forget() {
+        return forgetter;
+    }
+
+
+    /**
+     * increment the in count by 1
+     */
+    public void in() {
+        in.increment();
+        dirty = true;
+    }
+
+    /**
+     * increment the in count
+     *
+     * @param count the number to increment by
+     */
+    public void in(long count) {
+        in.increment(count);
+        dirty = true;
+    }
+
+    /**
+     * increment the out count by 1
+     */
+    public void out() {
+        out.increment();
+        dirty = true;
+    }
+
+    /**
+     * increment the count
+     *
+     * @param count the number by which to increment by
+     */
+    public void out(long count) {
+        out.increment(count);
+        dirty = true;
+    }
+
+    /**
+     * Get a reference to associated snitch to get discrete metric values.
+     *
+     * @return the associate {@link Snitch}
+     */
+    public Snitch snitch() {
+        return snitch;
+    }
+
+    /**
+     * Add to the existing queue push duration
+     *
+     * @param durationToAdd the duration to add
+     */
+    public void queuePushDuration(long durationToAdd) {
+        queuePushDuration.increment(durationToAdd);
+        dirty = true;
+    }
+
+    /**
+     * The snitch for the {@link EventsWitness}. Allows to read discrete metrics values.
+     */
+    public static class Snitch {
```

Comment:
Lol, this reminds me of Al Pacino's Scent of a Woman scene -- But, not a snitch! https://www.youtube.com/watch?v=PeCTb_gFgfY